test/lit.cfg

```diff
@@ -919,8 +919,8 @@ if run_os in ('maccatalyst',):
   target_os_abi = 'macosx'
   target_os_is_maccatalyst = "TRUE"
   config.available_features.add("OS=ios")
-# macOS on ASi uses the stable ABI
-if run_os in ('macosx',) and run_cpu in ('arm64',):
+# macOS and simulators on ASi use the stable ABI
+if (run_os in ('macosx',) or run_environment in ('-simulator',)) and run_cpu in ('arm64',):
```

Comment:
Do we want to keep the `-` in the environment match string?

Reply:
That's a good point, did not notice that since I was more focused on tests.

Took this feedback in e84c5bf8d2269da04bd621922ebc79c3bfe1ff34, 3476f313be3b6a195df1dc457b6aef28c38591ac